applications/nrf_desktop/src/modules/hid_forward.c

```diff
@@ -207,46 +218,44 @@ static struct enqueued_report *get_enqueued_report(struct enqueued_reports *enqu
 }
 
 static void drop_enqueued_reports(struct enqueued_reports *enqueued_reports,
-				  uint8_t report_id)
+				  uint8_t irep_idx)
 {
-	__ASSERT_NO_MSG(report_id < ARRAY_SIZE(enqueued_reports->reports));
+	__ASSERT_NO_MSG(irep_idx < ARRAY_SIZE(enqueued_reports->reports));
 
-	while (is_report_enqueued(enqueued_reports, report_id)) {
+	while (is_report_enqueued(enqueued_reports, irep_idx)) {
 		struct enqueued_report *item;
 
-		item = get_enqueued_report(enqueued_reports, report_id);
+		item = get_enqueued_report(enqueued_reports, irep_idx);
 
 		k_free(item->report);
 		k_free(item);
 	}
-
-	__ASSERT_NO_MSG(enqueued_reports->reports[report_id].count == 0);
```

Comment:
We still could use this assertion to make sure that `count` is consistent with number of items in the list.

Reply:
There is no point - if the assertion is false the loop would never end as the condition checks count and asserts that list is indeed empty. This is why I have removed it.

Reply:
Ok